### ISSUE TITLE
--fps doesn't work on videos

### ISSUE THREAD

**Issue by aikinai** [ImageOptim/gifski#23]
If I understand right, I think `--fps` just tells gifski the framerate to use when importing PNGs. It seems to have no effect on video. So if I have a 60fps video (which doesn't seem possible to play in any browser I've tested), I can't process video directly with gifski. There's also no option to downsample or ignore frames, so it seems like only option is to do the two-step (video→PNG→GIF) process, pass `-r` to ffmpeg to lower the framerate, and then pass that same framerate to gifski with `--fps`.

Would it be possible to support `--fps` on video so you can use the one-step process with high-framerate videos and downsample in gifski?

**Comment by jalba69** [comment]
Hello, 
maybe it is possible for now to use the ffmpeg pipe function to feed directly gifski without the need to write the png's

I don't recall the exact command for piping, nor i can test this since i don't have a gifski compiled with ffmpeg for using videos as input but should be something more or less like this : 
`ffmpeg -i input.mp4 -r 25 -f matroska pipe:1 |  gifski -o anim.gif  -i - `
Code needs a review as i don't recall exactly all but should be not much different.

You need to check for the exact piping commands and do some tests but might be possible this way, no guarantee tho.

**Comment by kornelski** [comment]
You're right that the `fps` doesn't work on videos. There's no framerate change option.

Unfortunately gifski doesn't support stdin, so the pipe won't work either. You'll have to use `ffmpeg` command with `-r` and write PNG files.

**Comment by jalba69** [comment]
Arg, if it doesn't support piping either i guess there isn't much remaining to try.

Maybe if you are on windows plateform and have avisynth installed it is worth a try feeding it directly with an .avs script having the framerate change in the script, like this for the .avs script and and .avi video input ( different commands and plugins for other video extensions ) :
```
AviSource("H:\MyInputVideo.avi", Audio = False)
ChangeFPS(25)
```
then feed gifski compiled with ffmpeg with command : 

`gifski -o animated.gif avisynth_script.avs`

I don't even know if the tool supports it but ffmpeg does so there is a little chance to get this working, if i manage to put my hands on a gifski.exe supporting video inputs i'll give a try.

**Comment by aikinai** [comment]
Oh okay, thanks for the response. For now I can write my own wrapper script to handle the ffmpeg step and clean up the PNGs, but I think this would be a key feature to make gifski more user friendly. Everyone's going to be starting with video (not images), and many videos these days are over 30fps so will need to be downsampled first.

**Comment by kornelski** [comment]
Fixed in 0.10.1
